cfg/gnu.cfg

```diff
@@ -28,7 +28,7 @@
   <function name="bswap_16">
     <noreturn>false</noreturn>
     <use-retval/>
-    <returnValue type="uint16_t"/>
+    <returnValue type="uint16_t">(((uint16_t)arg1 &amp; 0xff00u) &gt;&gt; 8) | (((uint16_t)arg1 &amp; 0x00ffu) &lt;&lt; 8)</returnValue>
```

Comment:
I do not understand what the casts accomplish. Imagine arg1 is 0xffffffff .. then `arg1 & 0xff00` is 0xff00 and `(uint16_t)arg1 & 0xff00` is 0xff00..

Reply:
I was more thinking of e.g. floating point arguments (although I am not sure they should ever be given to `bswap`). Normally there is always the (implicit) conversion to the function argument, but this seems to not be modeled without the casts. For integer arguments I suppose it it fine as long as I use large enough type suffixes for the constants.

Reply:
ok I think that is a good point.